source/server/config_validation/admin.h

```diff
@@ -30,8 +32,15 @@ class ValidationAdmin : public Admin {
   void addListenerToHandler(Network::ConnectionHandler* handler) override;
   uint32_t concurrency() const override { return 1; }
 
+  // We want to implement the socket interface without implementing the http listener function.
+  // This is useful for TAP because it wants to emit warnings when the address type is UDS
+  void createHttpListenerSocket(Network::Address::InstanceConstSharedPtr address) {
+    socket_ = std::make_shared<Network::TcpListenSocket>(nullptr, address, nullptr);
+  }
+
 private:
   ConfigTrackerImpl config_tracker_;
+  Network::SocketSharedPtr socket_;
```

Comment:
Can we have the `ValidationAdmin` take the config as part of its ctor and initialize this there instead?

Reply:
No strong opinion, here. I will create a constructor and add it there. Admin was default constructed and I wanted to keep it that way.

Reply:
Fixed in 2cf43a55597b14c77e62892549b89b4132c6dca9